excellent/scanner.go

```diff
@@ -9,11 +9,12 @@ import (
 	"unicode"
 )
 
-type xToken int
+// XTokenType is a set of types than can be scanned
+type XTokenType int
```

Comment:
go wasn't happy about some of these things being unexported but returned from exported functions